tx/textual/valuerenderer/coins.go

```diff
@@ -0,0 +1,171 @@
+package valuerenderer
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+
+	bankv1beta1 "cosmossdk.io/api/cosmos/bank/v1beta1"
+	basev1beta1 "cosmossdk.io/api/cosmos/base/v1beta1"
+	"cosmossdk.io/math"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+// NewCoinsValueRenderer returns a ValueRenderer for SDK Coin and Coins.
+func NewCoinsValueRenderer(q CoinMetadataQueryFn) ValueRenderer {
+	return coinsValueRenderer{q}
+}
+
+type coinsValueRenderer struct {
+	// coinMetadataQuerier defines a function to query the coin metadata from
+	// state.
```

Comment:
Add a note that this function is provided for testing convenience - in production always use the standard query function.

Reply:
This is not only used for testing. In production, this should be set to the bank keeper's query function.